internal/pkg/skuba/cni/cilium.go

```diff
@@ -104,32 +112,66 @@ func CiliumSecretExists(client clientset.Interface) (bool, error) {
 	return kubernetes.DoesResourceExistWithError(err)
 }
 
-func CreateOrUpdateCiliumConfigMap(client clientset.Interface) error {
-	etcdEndpoints := []string{}
-	apiEndpoints, err := kubeadm.GetAPIEndpointsFromConfigMap(client)
+// NeedsEtcdToCrdMigration checks if the migration from etcd to CRD is needed,
+// which is the case when upgrading from Cilium 1.5 to Cilium 1.6. Decision
+// depends on the old Cilium ConfigMap. If that config map exists and contains
+// the etcd config, migration has to be done.
```

Comment:
What if not? I think it might be good to explain why it's a noop when it's not the case... Future me will not remember the impact on not upgrading, and will prefer not to have to go through the PRs to find out.

Reply:
If not, the migration is just not done and Cilium without etcd config is deployed and that fresh Cilium deployment. If there is no old config map with etcd-config, it means that Cilium 1.5 was not deployed.

Reply:
Got it. Can you say that "Greefield Cilium 1.6+ won't have the old Cilium ConfigMap" in the comment ? I find this clearer.